### PR TITLE
Enable GLRenderer.

### PR DESCRIPTION
**Pull request by suyambulingamrm**
This CL
i. Adds method to retrieve texture backed SkImage from bitmap pixels.
ii. Enable GL Renderer to composite bitmap raster data.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

### CODE REVIEW

File: cc/resources/display_resource_provider.cc [hunk 1]
```diff
@@ -28,6 +29,7 @@ DisplayResourceProvider::DisplayResourceProvider(
 DisplayResourceProvider::~DisplayResourceProvider() {
   while (!children_.empty())
     DestroyChildInternal(children_.begin(), FOR_SHUTDOWN);
+  resource_sk_image_.clear();
```
Comment (Sunwoo-Nam) [inline cc/resources/display_resource_provider.cc]: How do you think whether a Castanets flag is needed for every modification or not?

File: components/viz/service/display/gl_renderer.cc [hunk 2]
```diff
@@ -600,7 +600,9 @@ void GLRenderer::DoDrawQuad(const DrawQuad* quad,
       DrawSolidColorQuad(SolidColorDrawQuad::MaterialCast(quad), clip_region);
       break;
     case DrawQuad::STREAM_VIDEO_CONTENT:
+#if !defined(CASTANETS)
       DrawStreamVideoQuad(StreamVideoDrawQuad::MaterialCast(quad), clip_region);
```
Comment (Sunwoo-Nam) [inline components/viz/service/display/gl_renderer.cc]: Why did you ignore video quad rendering?

File: cc/resources/display_resource_provider.cc (hunk 1 above)
Comment (suyambulingamrm) [reply cc/resources/display_resource_provider.cc]: I thought, the changes in ScopedReadLockSkImage is generic. So didnt add build flag.
I have added build flags accordingly in latest patch set.

File: components/viz/service/display/gl_renderer.cc (hunk 2 above)
Comment (suyambulingamrm) [reply components/viz/service/display/gl_renderer.cc]: Currently, StreamVideoQuad and YUV quads are not encountered in the supported scenarios.
Also it needs some verification to make the changes for these quads.
So I have added a TODO for now. Once these quads are supported/needed we will add a fixup.
Should that be fine ?

**Pull request merged**